lumigator/schemas/lumigator_schemas/jobs.py

```diff
@@ -151,10 +151,37 @@ class JobInferenceConfig(BaseJobConfig):
     )
 
 
-class JobAnnotateConfig(BaseJobConfig):
-    job_type: Literal[JobType.ANNOTATION] = JobType.ANNOTATION
-    task: TaskType = Field(default=TaskType.SUMMARIZATION)
-    store_to_dataset: bool = False
+class JobAnnotateConfig(JobInferenceConfig):
+    """Job configuration for the annotation job type
+
+    An annotation job is a special type of inference job that is used to
+    annotate a dataset with predictions from a model. The predictions are
+    stored in the dataset as a new field called `ground_truth`.
+
+    JobAnnotateConfig inherits from JobInferenceConfig but fixes the following
+    fields, using the `SkipJsonSchema` type to prevent them from being included
+    in the JSON schema:
+    - job_type: Literal[JobType.ANNOTATION]
+    - output_field: "ground_truth"
+
+    It also sets sensible defaults for the following fields:
+    - store_to_dataset: True
+    - model: "facebook/bart-large-cnn"
+    - provider: "hf"
+
+    Users can change the model and provider fields but cannot change the
+    job_type or output_field fields.
+
+    Note that, currently, ground truth generation is limited to summarization
+    tasks from the UI. Users can run any ground truth generation task from the
+    API.
+    """
+
+    job_type: SkipJsonSchema[Literal[JobType.ANNOTATION]] = JobType.ANNOTATION
```

Comment:
If we take the approach outlined above by Irina, I'd also avoid modifying the schema, since it will be the job's responsibility to adapt the incoming config to the config used by the job.

An alternative is to use a redefinition of the Inference config that uses optional, literal fields https://docs.pydantic.dev/1.10/usage/types/#literal-type

In this way, the user doesn't specify a value, so it takes the default value; and the user _cannot_ explicitly specify another value than the default one.

The SkipJsonSchema option feels fairly hackish to me. Even if it's not present in the schema, we should verify whether the code may still build the entity from the existing json data.

Reply:
Let's do it like this for the moment, we can open this discussion in a later PR.